neurolib/models/model.py

```diff
@@ -2,34 +2,321 @@
 import xarray as xr
 import numpy as np
 
+from neurolib.models import bold
+
 from neurolib.utils.collections import dotdict
 
 
 class Model:
     """The Model superclass manages inputs and outputs of all models.
     """
 
-    defaultOutput = None
+    def __init__(
+        self,
+        integration,
+        params,
+        state_vars,
+        init_vars=None,
+        output_vars=None,
+        input_vars=None,
+        default_output=None,
+        bold=False,
+        normalize_bold_input=False,
+        normalize_bold_input_max=50,
+        name=None,
+        description=None,
+    ):
+        if name is not None:
+            assert isinstance(name, str), f"Model name is not a string."
+            self.name = name
+
+        assert integration is not None, "Model integration function not given."
+        self.integration = integration
+
+        assert isinstance(params, dict), "Parameters must be a dictionary."
+        self.params = dotdict(params)
 
-    def __init__(self, name, description=None):
-        assert isinstance(name, str), f"Model name is not a string."
-        self.name = name
+        self.state_vars = state_vars
```

Comment:
as I was saying, when in child class you are setting this outside the scope of `__init__` as attributes, you do not have to assign them to `self` like here...